euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Collector.java

```diff
@@ -29,4 +29,13 @@
    * @param elem the element to collect
    */
   void collect(T elem);
+
+  /**
+   * Returns {@link Context} view of the collector.
+   * Since {@link Collector} usually share the same methods
+   * as {@link Context} it can be safely casted.
+   *
+   * @return this instance as a context class
+   */
+  Context asContext();
```

Comment:
It is not possible to directly cast `Collector` to `Context` since they are not subtypes of each other. The implementation of this doesn't rely on casting, see https://github.com/seznam/euphoria/pull/132/files#diff-e704b8f3c221db98ac7f7a4618484531R63